src/NServiceBus.Core/Transports/Msmq/MsmqTransportInfrastructure.cs

```diff
@@ -33,17 +33,19 @@ public MsmqTransportInfrastructure(ReadOnlySettings settings, string connectionS
 
         ReceiveStrategy SelectReceiveStrategy(TransportTransactionMode minimumConsistencyGuarantee, TransactionOptions transactionOptions)
         {
-            if (minimumConsistencyGuarantee == TransportTransactionMode.TransactionScope)
+            switch (minimumConsistencyGuarantee)
             {
-                return new ReceiveWithTransactionScope(transactionOptions, new MsmqFailureInfoStorage(1000));
+                case TransportTransactionMode.TransactionScope:
+                    return new ReceiveWithTransactionScope(transactionOptions, new MsmqFailureInfoStorage(1000));
```

Comment:
Was picked by recoverability and previous TFs as a reasonable nummer for failure InMem storage. Not exposed as config for now. Out of scope for this PR